runtime/lua/vim/lsp/handlers.lua

```diff
@@ -512,6 +512,52 @@ M['window/showMessage'] = function(_, result, ctx, _)
   return result
 end
 
+--see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#window_showDocument
+M['window/showDocument'] = function(_, result, ctx, _)
+  local uri = result.uri
+
+  if result.external then
+    -- TODO(lvimuser): ask the user for confirmation
+    local cmd
+    if vim.fn.has('win32') == 1 then
+      cmd = { 'cmd.exe', '/c', 'start', '""', vim.fn.shellescape(uri) }
+    elseif vim.fn.has('macunix') == 1 then
+      cmd = { 'open', vim.fn.shellescape(uri) }
+    else
+      cmd = { 'xdg-open', vim.fn.shellescape(uri) }
+    end
+
+    local ret = vim.fn.system(cmd)
+    if vim.v.shellerror ~= 0 then
```

Comment:
Variable name is wrong, so will always show an error. Should be `vim.v.shell_error`